controllers/slicegateway/reconciler.go

```diff
@@ -192,8 +192,75 @@ func (r *SliceGwReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 			)
 			return ctrl.Result{}, err
 		}
+		//reconcile the available nodes
+		if err := r.reconcileNodes(ctx, sliceGw); err != nil {
+			return ctrl.Result{}, err
+		}
 	}
 
+	isClient := sliceGw.Status.Config.SliceGatewayHostType == "Client"
+	if isClient {
+		//check if SliceGatewayRemoteGatewayID is populated , else requeue
+		if sliceGw.Status.Config.SliceGatewayRemoteNodeIP == "" || sliceGw.Status.Config.SliceGatewayRemoteGatewayID == "" {
```

Comment:
This check should be avoided, instead readyToDeployGwClient() should be enhanced.